File: kotlin/src/Programmers/두큐합같게만들기.kt
// https://school.programmers.co.kr/learn/courses/30/lessons/118667

class Solution {
    fun solution(queue11: IntArray, queue22: IntArray): Int {
        var answer: Int = 0
        val max_cnt: Int = (queue11.size + queue22.size)*2

        var queue1 = ArrayDeque<Long>()
        var queue2 = ArrayDeque<Long>()
        for (i in queue11.indices) {
            queue1.add(queue11[i].toLong())
            queue2.add(queue22[i].toLong())
        }

        var sum1: Long = queue1.sum()
        var sum2: Long = queue2.sum()

        while(sum1 != sum2){
            if (sum1 >= sum2){
                var element = queue1.removeFirst()
                sum1 -= element
                sum2 += element
                queue2 += element
                answer += 1
            }else{
                var element = queue2.removeFirst()
                sum2 -= element
                sum1 += element
                queue1 += element
                answer += 1
            }
            if (answer > max_cnt){
                answer = -1
                break
            }
        }

        return answer
    }
}

fun main(){
    val solution = Solution()
    println(solution.solution(intArrayOf(3, 2, 7, 2), intArrayOf(4, 6, 5, 1)))
}

File: python/Programmers/두큐합같게만들기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/118667

from collections import deque

def solution(queue1, queue2):
    max_cnt = (len(queue1)+len(queue2))*2
    queue1 = deque(queue1)
    queue2 = deque(queue2)
    
    # 시간초과 원인
    sum1 = sum(queue1)
    sum2 = sum(queue2)

    answer = 0
    while sum1 != sum2:
        if sum1 >= sum2:
            element = queue1.popleft()
            queue2.append(element)
            answer += 1
            sum1 -= element
            sum2 += element
        else:
            element = queue2.popleft()
            queue1.append(element)
            answer += 1
            sum2 -= element
            sum1 += element
        if answer > max_cnt:
            answer = -1
            break
    
    return answer